# EX3: Some Statistics with Baby Names

### Introduction:

We are going to use a subset of [US Baby Names](https://www.kaggle.com/kaggle/us-baby-names) from Kaggle.  
In the file it will be names from 2004 until 2014


### Step 1. Import the necessary libraries

In [1]:
#import pandas as pd

import numpy as np
import pandas as pd

### Step 2. 
User ***Jupyter Lab*** to view ``../../DATA/US_Baby_Names_right.csv'``.

### Step 3. Assign it to a variable called baby_names.

In [2]:
baby_names=pd.read_csv('../../DATA/US_Baby_Names_right.csv')

### Step 4. See the first 10 entries

In [3]:
baby_names.head(10) #Erste 10 Einträge

Unnamed: 0     Id      Name  Year Gender State  Count
0       11349  11350      Emma  2004      F    AK     62
1       11350  11351   Madison  2004      F    AK     48
2       11351  11352    Hannah  2004      F    AK     46
3       11352  11353     Grace  2004      F    AK     44
4       11353  11354     Emily  2004      F    AK     41
5       11354  11355   Abigail  2004      F    AK     37
6       11355  11356    Olivia  2004      F    AK     33
7       11356  11357  Isabella  2004      F    AK     30
8       11357  11358    Alyssa  2004      F    AK     29
9       11358  11359    Sophia  2004      F    AK     28

### Step 5. Delete the column 'Unnamed: 0' and 'Id'

In [17]:
baby_names_adjusted = baby_names.drop(columns=['Unnamed: 0', 'Id']) #Frage: Warum wird die Methode drop nicht auf den Datensatz angewendet?

In [13]:
baby_names_adjusted

Name  Year Gender State  Count
0           Emma  2004      F    AK     62
1        Madison  2004      F    AK     48
2         Hannah  2004      F    AK     46
3          Grace  2004      F    AK     44
4          Emily  2004      F    AK     41
...          ...   ...    ...   ...    ...
1016390     Seth  2014      M    WY      5
1016391  Spencer  2014      M    WY      5
1016392     Tyce  2014      M    WY      5
1016393   Victor  2014      M    WY      5
1016394   Waylon  2014      M    WY      5

[1016395 rows x 5 columns]

### Step 6. Are there more male or female names in the dataset?

In [15]:
#Anzahl Rows in denen Gender = F
len(baby_names_adjusted[baby_names_adjusted['Gender'] == 'F'])

558846

In [14]:
#Anzahl Rows in denen Gender = M
len(baby_names_adjusted[baby_names_adjusted['Gender'] == 'M'])

457549

Antwort: Es sind mehr female names im dataset.

### Step 7. Group the dataset by name and assign to names

In [16]:
#group by name 
baby_namesByName = baby_names_adjusted.groupby(baby_names_adjusted['Name'])
baby_namesByName.count() #number of group entries by columns

Year  Gender  State  Count
Name                               
Aaban       2       2      2      2
Aadan       4       4      4      4
Aadarsh     1       1      1      1
Aaden     196     196    196    196
Aadhav      1       1      1      1
...       ...     ...    ...    ...
Zyra        7       7      7      7
Zyrah       2       2      2      2
Zyren       1       1      1      1
Zyria      10      10     10     10
Zyriah      9       9      9      9

[17632 rows x 4 columns]

### Step 8. How many different names exist in the dataset?

In [24]:
baby_namesByName.ngroups

17632

Antwort: Im Datensatz existieren 17632 Namen

### Step 9. What is the name with most occurrences?

In [69]:
#baby_namesByName[baby_namesByName['Count'].sum().max()]

In [70]:
baby_namesByName['Count'].sum()

Name
Aaban        12
Aadan        23
Aadarsh       5
Aaden      3426
Aadhav        6
           ... 
Zyra         42
Zyrah        11
Zyren         6
Zyria        59
Zyriah       58
Name: Count, Length: 17632, dtype: int64

In [71]:
baby_namesByName['Count'].sum().max()

242874

In [68]:
baby_namesByName

### Step 10. How many different names have the least occurrences?

### Step 11. What is the median name occurrence?

### Step 12. What is the standard deviation of names?

### Step 13. Get a summary with the mean, min, max, std and quartiles.